### ISSUE TITLE
Failing tests on update_to_90

### ISSUE THREAD

**Issue by rgal** [SNSystems/llvm-project-prepo#79]
`update_to_90` branch merges from LLVM 9.0 branch point.

`repo_pruning_dependents` tests are failing on that branch.

Tested on Ubuntu 18.04, pstore revision 30f09234825e40ea860626a513b6d17326860ed2:

```
$ python bin/llvm-lit -v ../llvm/test/Feature/Repo/repo_pruning_dependents*.ll
-- Testing: 4 tests, 4 threads --
FAIL: LLVM :: Feature/Repo/repo_pruning_dependents_3.ll (1 of 4)
******************** TEST 'LLVM :: Feature/Repo/repo_pruning_dependents_3.ll' FAILED ********************
Script:
--
: 'RUN: at line 7';   rm -f /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_3.ll.tmp.db
: 'RUN: at line 8';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_3.ll.tmp.db /home/russell/git/llvm-project-prepo/stage1/bin/llc -filetype=obj /home/russell/git/llvm-project-prepo/llvm/test/Feature/Repo/Inputs/repo_pruning_dependent_with_private_linkage.ll -o /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_3.ll.tmp
: 'RUN: at line 9';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_3.ll.tmp.db /home/russell/git/llvm-project-prepo/stage1/bin/llc -filetype=obj /home/russell/git/llvm-project-prepo/llvm/test/Feature/Repo/repo_pruning_dependents_3.ll -o /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_3.ll.tmp1
: 'RUN: at line 10';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_3.ll.tmp.db pstore-dump --all-compilations /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_3.ll.tmp.db | /home/russell/git/llvm-project-prepo/stage1/bin/FileCheck /home/russell/git/llvm-project-prepo/llvm/test/Feature/Repo/repo_pruning_dependents_3.ll
--
Exit Code: 1

Command Output (stderr):
--
LLVM ERROR: The digest of missing repository fragment 019f6773fcfe09627c507f531577f3a7 was found in a compilation member.

--

********************
FAIL: LLVM :: Feature/Repo/repo_pruning_dependents_2.ll (2 of 4)
******************** TEST 'LLVM :: Feature/Repo/repo_pruning_dependents_2.ll' FAILED ********************
Script:
--
: 'RUN: at line 3';   rm -f /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_2.ll.tmp.db
: 'RUN: at line 4';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_2.ll.tmp.db /home/russell/git/llvm-project-prepo/stage1/bin/llc -filetype=obj /home/russell/git/llvm-project-prepo/llvm/test/Feature/Repo/Inputs/repo_pruning_dependent.ll -o /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_2.ll.tmp
: 'RUN: at line 5';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_2.ll.tmp.db pstore-dump  --all-compilations /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_2.ll.tmp.db > /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_2.ll.tmp.log
: 'RUN: at line 6';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_2.ll.tmp.db /home/russell/git/llvm-project-prepo/stage1/bin/llc -filetype=obj /home/russell/git/llvm-project-prepo/llvm/test/Feature/Repo/repo_pruning_dependents_2.ll -o /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_2.ll.tmp1
: 'RUN: at line 7';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_2.ll.tmp.db pstore-dump  --all-compilations /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_2.ll.tmp.db > /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_2.ll.tmp1.log
: 'RUN: at line 8';   diff /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_2.ll.tmp.log /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_2.ll.tmp1.log
--
Exit Code: 1

Command Output (stderr):
--
LLVM ERROR: The digest of missing repository fragment 1a907d11b14d9f2a556c4bd21d0f2334 was found in a compilation member.

--

********************
FAIL: LLVM :: Feature/Repo/repo_pruning_dependents.ll (3 of 4)
******************** TEST 'LLVM :: Feature/Repo/repo_pruning_dependents.ll' FAILED ********************
Script:
--
: 'RUN: at line 16';   rm -f /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents.ll.tmp.db
: 'RUN: at line 17';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents.ll.tmp.db /home/russell/git/llvm-project-prepo/stage1/bin/llc -mtriple="x86_64-pc-linux-gnu-repo" -filetype=obj /home/russell/git/llvm-project-prepo/llvm/test/Feature/Repo/repo_pruning_dependents.ll -o /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents.ll.tmp
: 'RUN: at line 18';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents.ll.tmp.db pstore-dump  --all-compilations /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents.ll.tmp.db > /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents.ll.tmp.log
: 'RUN: at line 19';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents.ll.tmp.db clang -O3 -c --target=x86_64-pc-linux-gnu-repo -x ir /home/russell/git/llvm-project-prepo/llvm/test/Feature/Repo/repo_pruning_dependents.ll  -o /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents.ll.tmp1
: 'RUN: at line 20';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents.ll.tmp.db pstore-dump  --all-compilations /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents.ll.tmp.db > /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents.ll.tmp1.log
: 'RUN: at line 21';   diff /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents.ll.tmp.log /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents.ll.tmp1.log
--
Exit Code: 1

Command Output (stdout):
--
26a27,36
>       - digest      : e275cedb0b4638bb3ffaf92cf262a1a6
>         compilation :
>             members :
>                 - digest     : e7a85a8e1b446bbc277d1344b12334fa
>                   fext       : { addr: 1280, size: 46 }
>                   name       : A
>                   linkage    : external
>                   visibility : default
>             path    : /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output
>             triple  : x86_64-pc-linux-gnu-repo

--

********************
FAIL: LLVM :: Feature/Repo/repo_pruning_dependents_1.ll (4 of 4)
******************** TEST 'LLVM :: Feature/Repo/repo_pruning_dependents_1.ll' FAILED ********************
Script:
--
: 'RUN: at line 17';   rm -f /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_1.ll.tmp.db
: 'RUN: at line 18';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_1.ll.tmp.db /home/russell/git/llvm-project-prepo/stage1/bin/llc -mtriple="x86_64-pc-linux-gnu-repo" -filetype=obj /home/russell/git/llvm-project-prepo/llvm/test/Feature/Repo/repo_pruning_dependents_1.ll -o /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_1.ll.tmp
: 'RUN: at line 19';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_1.ll.tmp.db pstore-dump  --all-compilations /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_1.ll.tmp.db > /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_1.ll.tmp.log
: 'RUN: at line 20';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_1.ll.tmp.db clang -O3 -c --target=x86_64-pc-linux-gnu-repo -x ir /home/russell/git/llvm-project-prepo/llvm/test/Feature/Repo/repo_pruning_dependents_1.ll  -o /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_1.ll.tmp1
: 'RUN: at line 21';   env REPOFILE=/home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_1.ll.tmp.db pstore-dump  --all-compilations /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_1.ll.tmp.db > /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_1.ll.tmp1.log
: 'RUN: at line 22';   diff /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_1.ll.tmp.log /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output/repo_pruning_dependents_1.ll.tmp1.log
--
Exit Code: 1

Command Output (stdout):
--
6a7,21
>       - digest      : 8961cc5e8054400f62bcc05fb0b3d5ee
>         compilation :
>             members :
>                 - digest     : a539a575a5c29f141e5e189642725a3a
>                   fext       : { addr: 1328, size: 46 }
>                   name       : A
>                   linkage    : external
>                   visibility : default
>                 - digest     : ce00a69aa25c8d21907b3ed3a2fd5033
>                   fext       : { addr: 1240, size: 46 }
>                   name       : B
>                   linkage    : external
>                   visibility : default
>             path    : /home/russell/git/llvm-project-prepo/stage1/test/Feature/Repo/Output
>             triple  : x86_64-pc-linux-gnu-repo

--

********************
Testing Time: 0.56s
********************
Failing Tests (4):
    LLVM :: Feature/Repo/repo_pruning_dependents.ll
    LLVM :: Feature/Repo/repo_pruning_dependents_1.ll
    LLVM :: Feature/Repo/repo_pruning_dependents_2.ll
    LLVM :: Feature/Repo/repo_pruning_dependents_3.ll

  Unexpected Failures: 4
```

**Comment by MaggieYingYi** [comment]
I am trying to reproduce the test failures. I used the same pstore commit you mentioned. However, I got two test failures instead of four test failures. 

```
Failing Tests (2):
    LLVM :: Feature/Repo/repo_pruning_dependents_2.ll
    LLVM :: Feature/Repo/repo_pruning_dependents_3.ll

  Expected Passes    : 34428
  Expected Failures  : 68
  Unsupported Tests  : 14431
  Unexpected Failures: 2
```

I built the project with Release configure. @rgal , Did you build with Debug or Release configuration?

In addition, I found the llvm-project-prepo can't build on Windows with VS2017 with the following error:

```
184>    c:\mywork\git\github\llvm-project-prepo\llvm\lib\mc\mcrepoticketfile.cpp(111): error C2664: 'llvm::ErrorOr<std::unique_ptr<llvm::MemoryBuffer,std::default_delete<_Ty>>> llvm::MemoryBuffer::getOpenFile(llvm::sys::fs::file_t,const llvm::Twine &,uint64_t,bool,bool)': cannot convert argument 1 from 'int' to 'llvm::sys::fs::file_t'
184>            with
184>            [
184>                _Ty=llvm::MemoryBuffer
184>            ]
```
@rgal , Did you see this error when build the project on Windows with Debug configuration?

**Comment by rgal** [comment]
I was using Release with asserts enabled.

I've been working on Linux for this rather than Windows. I don't think I changed that in the merge so I think that would be there from before this.

**Comment by MaggieYingYi** [comment]
> I was using Release with asserts enabled.

Thanks Russ, after enabling asserts, I get the same four test failures.

**Comment by MaggieYingYi** [comment]
Fixed in commit 5a5545a.
